src/Control/HTTPApplication.php

```diff
@@ -141,4 +152,85 @@ public function execute(HTTPRequest $request, callable $callback, $flush = false
             $this->getKernel()->shutdown();
         }
     }
+
+    private function warnAboutDeprecatedSetups()
+    {
+        // TypeCreator is a class unique to GraphQL v3 - we use it in other areas to detect
+        // which version is being used.
+        if (class_exists(TypeCreator::class)) {
+            Deprecation::notice(
+                '4.13.0',
+                'silverstripe/graphql 3.x is deprecated. Upgrade to 4.x instead.'
+                . ' See https://docs.silverstripe.org/en/4/upgrading/upgrading_to_graphql_4/',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // The alternate_base_folder config property is deprecated, but because it's
+        // always fetched it'll throw a deprecation warning whether you've set it or not.
+        if (Director::baseFolder() !== BASE_PATH) {
+            Deprecation::notice(
+                '4.13.0',
+                'Use of an alternate base folder is deprecated',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
```

Comment:
I'm.... not actually sure why this one is deprecated? It's not deprecated _or removed_ in CMS 5 so.... don't merge until we've sorted that out I guess.

Reply:
From Steve on slack:
> Looks like a mistake, probably mistook it for `alternate_public_dir` which was removed

I've undeprecated that config now, looks like it shouldn't have been deprecated in the first place.